client/driver/docker.go

```diff
@@ -166,6 +167,32 @@ func (d *DockerDriver) createContainer(ctx *ExecContext, task *structs.Task) (do
 	d.logger.Printf("[DEBUG] driver.docker: using %d cpu shares for %s", hostConfig.CPUShares, task.Config["image"])
 	d.logger.Printf("[DEBUG] driver.docker: binding directories %#v for %s", hostConfig.Binds, task.Config["image"])
 
+	//  set privileged (fallback to false)
+	hostConfig.Privileged, _ = strconv.ParseBool(task.Config["privileged"])
```

Comment:
Since privileged is a potentially dangerous flag, we want to give the Nomad operator control over whether this is allowed. So what should happen is there should be a name key/value to enable or disable the docker driver from allowing the privileged flag:

That would go here: https://github.com/hashicorp/nomad/blob/master/client/config/config.go#L55
Here is an example: https://github.com/hashicorp/nomad/blob/cb811dd97ccd81b56591737f7d9fa1b445cda095/client/driver/docker.go#L76

And the default should be that privileged is not enabled. The docker driver should also set a node attribute to indicate if it allows privileged. Let me know if you need clarification